compiler/src/dotty/tools/dotc/transform/InstrumentCoverage.scala

```diff
@@ -331,6 +343,20 @@ class InstrumentCoverage extends MacroTransform with IdentityDenotTransformer:
 
       cpy.CaseDef(tree)(pat, transformedGuard, instrumentedBody)
 
+    /** Transforms the parents of a Template. */
+    private def transformTemplateParents(parents: List[Tree])(using Context): List[Tree] =
+      def transformParent(parent: Tree): Tree = parent match
+        case tree: Apply =>
+          // only instrument the args, not the constructor call
+          cpy.Apply(tree)(tree.fun, tree.args.mapConserve(transform))
+        case tree: TypeApply =>
+          // args are types, instrument the fun with transformParent
+          cpy.TypeApply(tree)(transformParent(tree.fun), tree.args)
+        case other =>
+          transform(other)
```

Comment:
Does the transform here do anything or could we just return `other`? I think other here would always be a TypeTree so no instrumentatin will be added.

Reply:
I think you're right, yes